signature/cose/envelope.go

```diff
@@ -569,3 +596,55 @@ func contains(s []interface{}, e interface{}) bool {
 	}
 	return false
 }
+
+// encodeTime generates a Tag1 Datetime CBOR object and casts it to
+// cbor.RawMessage
+func encodeTime(t time.Time) (cbor.RawMessage, error) {
+	encMode, err := encOpts.EncMode()
+	if err != nil {
+		return nil, err
+	}
+
+	timeCBOR, err := encMode.Marshal(t)
+	if err != nil {
+		return nil, err
+	}
+
+	return cbor.RawMessage(timeCBOR), nil
+}
+
+// decodeTime decodes cbor.RawMessage of Tag1 Datetime CBOR object
+// into time.Time
+func decodeTime(timeRaw cbor.RawMessage) (time.Time, error) {
+	decMode, err := decOpts.DecMode()
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	var t time.Time
+	err = decMode.Unmarshal([]byte(timeRaw), &t)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return t, nil
+}
+
+// parseTime parses time values from cose.ProtectedHeader
+// in go-cose, CBOR times (tag 1) decode to time.Time.
+//
+// For more details: https://github.com/fxamacker/cbor/blob/7704fa5efaf3ef4ac35aff38f50f6ff567793072/decode.go#L52
+func parseTime(timeValue interface{}) (time.Time, error) {
+	switch t := timeValue.(type) {
+	case cbor.RawMessage:
+		decTime, err := decodeTime(t)
+		if err != nil {
+			return time.Time{}, err
+		}
+		return decTime, nil
+	case time.Time:
```

Comment:
There is issue here about the `time.Time`. How do we ensure it is parsed from tag `1` instead of tag `0`?

Reply:
created an issue to track this.